lib/howl/commands/app_commands.moon

```diff
@@ -665,39 +668,112 @@ do_search = (search, whole_word) ->
       m = matches[i]
       file_search_hit_to_location(m, search, display_as)
 
-  locations, searcher, project
+  locations, searcher
 
-command.register
-  name: 'project-file-search',
-  description: 'Searches files in the the current project'
-  input: (opts) ->
-    editor = app.editor
-    search = nil
-    whole_word = false
+ask_for_search_directory = (search_term) ->
+  start_dir = get_buffer_dir(app.editor.buffer)
+  if not start_dir
+    start_dir = howl.io.File(howl.sys.env.HOME)
 
-    if opts.text and not opts.text.is_empty
-      search = opts.text
+  title = if search_term
+    "(Please specify a directory to search for '#{search_term}'): "
+  else
+    "(Please specify a directory to search): "
 
-    unless search or app.window.command_panel.is_active
-      if editor.selection.empty
-        search = app.editor.current_context.word.text
-        whole_word = true unless search.is_empty
-      else
-        search = editor.selection.text
+  directory = interact.select_directory
+    title: title
+    prompt: "Directory: "
+    path: start_dir.path
+
+  directory
+
+-- Common input handler for search commands displaying results
+-- in a pop-up panel. E.g. project-file-search and folder-search
+search_input = (opts, get_directory) ->
```

Comment:
Instead of passing a function `get_directory`, can we just pass `opts.directory`? If nil, this can invoke `ask_for_search_directory` but if present, it just gets used? Then the caller from the project commands can just pass the directory instead of a function.